openpmd_validator/check_h5.py

```diff
@@ -702,20 +714,38 @@ def check_particles(f, iteration, v, extensionStates) :
     result_array = np.array([ 0, 0]) 
 
     # Find the path to the data
-    base_path = ("/data/%s/" % iteration).encode('ascii')
+    base_path = "/data/%s/" % iteration
     valid, particles_path = get_attr(f, "particlesPath")
-    if os.path.join( base_path, particles_path) !=  \
-        ( base_path + particles_path ) :
-        print("Error: `basePath`+`particlesPath` seems to be malformed "
-            "(is `basePath` absolute and ends on a `/` ?)")
-        return( np.array([1, 0]) )
+
+    if not valid and v:
+        print("`particlesPath` attribute is missing in '/' "
+              "(will not search for particle records)")
+        particles_path = None
```

Comment:
I think that this should be:
```python
if not valid and v:
    particles_path = None
    if v:
        print("`particlesPath` attribute is missing in '/' "
              "(will not search for particle records)")
else:
    particles_path = particles_path.decode()
```
(Same thing with `meshes_path`)

Otherwise this raises an error when trying to `decode` `None`.

Reply:
oh yes!